llvm/lib/Analysis/ValueTracking.cpp

```diff
@@ -8285,6 +8285,15 @@ static std::optional<bool> isImpliedCondICmps(const ICmpInst *LHS,
   if (areMatchingOperands(L0, L1, R0, R1, AreSwappedOps))
     return isImpliedCondMatchingOperands(LPred, RPred, AreSwappedOps);
 
+  if (RPred == ICmpInst::ICMP_SLT) {
+    Value *X;
+    Value *Y;
+    // x-y+1 is positive when x >= y or non-positive when x < y
```

Comment:
```suggestion
    // x -nsw y is non-negative when x >= y or negative when x < y
```
You match only the sub here, not an extra +1, so the comment should match that.

Reply:
hi, @nikic,
   the **extra +1** is matched with the **match(R1, m_AllOnes())**.